ns1/resource_record.go

```diff
@@ -514,8 +514,14 @@ func metaDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
 
 func metaDiffSuppressUp(k, old, new string, _ *schema.ResourceData) bool {
 	if strings.HasSuffix(k, "up") {
-		newB, _ := strconv.ParseBool(new)
-		oldB, _ := strconv.ParseBool(old)
+		newB, err := strconv.ParseBool(new)
+		if err != nil {
+			return new == old
+		}
+		oldB, err := strconv.ParseBool(old)
+		if err != nil {
+			return new == old
+		}
```

Comment:
on second thought, this seems crummy. not sure if the `new == old` comparisons would always be correct, or legal ...

Reply:
I think it's probably ok -- whatever is stored in state is a string so if the ParseBool fails its just a string comparison. Hopefully okay for most cases

Reply:
Why perform the comparison at all if there's an err?  Wouldn't it be simpler to just `return false` on err, since we only want to supress the diff for equal bool like values?

Reply:
my concern with just returning False on error would be identifying changes to the string value as "no change", but maybe I'm not fully grokking how the suppression funcs are used?

Reply:
diffSuppressFuncs are pretty poorly documented imo, but they control whether or not a detected diff should be suppressed, not the actual value of the diff.  The documentation on the code is way more clear here than hashi's official docs https://github.com/hashicorp/terraform-plugin-sdk/blob/9f0df37a8fdb2627ae32db6ceaf7f036d89b6768/helper/schema/schema.go#L263-L268

By returning `False` here, we're just saying if the value isn't a bool, any detected diff is valid and should be retained.

Reply:
word up looks legit